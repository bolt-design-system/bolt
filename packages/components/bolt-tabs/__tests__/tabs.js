import {
  render,
  renderString,
  stopServer,
  html,
  vrtDefaultConfig as vrtConfig,
  $RefParser,
} from '../../../testing/testing-helpers';
const { join } = require('path');
const schemaFilePath = join(__dirname, '../tabs.schema.yml');

const timeout = 120000;

describe('<bolt-tabs> Component', () => {
  let page, schema, props;

  beforeEach(async () => {
    await page.evaluate(() => {
      document.body.innerHTML = '';
    });
  }, timeout);

  beforeAll(async () => {
    schema = await $RefParser.dereference(schemaFilePath);
    props = schema.properties;
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
    });
  }, timeout);

  afterAll(async () => {
    await stopServer();
    await page.close();
  }, timeout);

  test('basic usage', async () => {
    const results = await render('@bolt-components-tabs/tabs.twig', {
      panels: [
        {
          label: 'Tab label 1',
          content: 'This is the tab content.',
        },
        {
          label: 'Tab label 2',
          content: 'This is the tab content.',
        },
        {
          label: 'Tab label 3',
          content: 'This is the tab content.',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // todo: why isn't this working?
  // $RefParser.dereference(schemaFilePath, (err, schema) => {
  //   schema.properties.spacing.enum.forEach(async spacingChoice => {
  //     test(`spacing: ${spacingChoice}`, async () => {
  //       const results = await render('@bolt-components-tabs/tabs.twig', {
  //         spacing: spacingChoice,
  //         panels: [
  //           {
  //             label: 'Tab label 1',
  //             content: 'This is the tab content.',
  //           },
  //           {
  //             label: 'Tab label 2',
  //             content: 'This is the tab content.',
  //           },
  //           {
  //             label: 'Tab label 3',
  //             content: 'This is the tab content.',
  //           },
  //         ],
  //       });
  //       expect(results.ok).toBe(true);
  //       expect(results.html).toMatchSnapshot();
  //     });
  //   });
  // });
});
