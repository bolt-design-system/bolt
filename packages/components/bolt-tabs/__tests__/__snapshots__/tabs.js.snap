// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Tabs Align: center 2`] = `
<bolt-tabs
  align="center"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Align: end 2`] = `
<bolt-tabs
  align="end"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Align: start 2`] = `
<bolt-tabs
  align="start"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Inset: auto 2`] = `
<bolt-tabs
  inset="auto"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Inset: off 2`] = `
<bolt-tabs
  inset="off"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Inset: on 2`] = `
<bolt-tabs
  inset="on"
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;

exports[`Bolt Tabs Twig usage 1`] = `
<bolt-tabs>
  <div class="c-bolt-tabs c-bolt-tabs--align-start c-bolt-tabs--inset">
    <input type="radio"
           name="tabs-12345"
           id="tabs-12345-1"
           class="c-bolt-tabs__input"
           checked
    >
    <label for="tabs-12345-1"
           class="c-bolt-tabs__label c-bolt-tabs__label--spacing-small"
    >
      <span class="c-bolt-tabs__label-inner">
        <span class="c-bolt-tabs__label-text">
          Tab label 1
        </span>
      </span>
    </label>
    <div class="c-bolt-tab-panel__outer">
      <ssr-keep for="bolt-tabs">
        <bolt-tab-panel>
          <ssr-keep for="bolt-tab-panel">
            <div slot="label">
              Tab label 1
            </div>
          </ssr-keep>
          <div class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small">
            <div role="tabpanel"
                 class="c-bolt-tab-panel__content"
            >
              <ssr-keep for="bolt-tab-panel">
                This is the tab content.
              </ssr-keep>
            </div>
          </div>
        </bolt-tab-panel>
      </ssr-keep>
    </div>
    <input type="radio"
           name="tabs-12345"
           id="tabs-12345-2"
           class="c-bolt-tabs__input"
    >
    <label for="tabs-12345-2"
           class="c-bolt-tabs__label c-bolt-tabs__label--spacing-small"
    >
      <span class="c-bolt-tabs__label-inner">
        <span class="c-bolt-tabs__label-text">
          Tab label 2
        </span>
      </span>
    </label>
    <div class="c-bolt-tab-panel__outer">
      <ssr-keep for="bolt-tabs">
        <bolt-tab-panel>
          <ssr-keep for="bolt-tab-panel">
            <div slot="label">
              Tab label 2
            </div>
          </ssr-keep>
          <div class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small">
            <div role="tabpanel"
                 class="c-bolt-tab-panel__content"
            >
              <ssr-keep for="bolt-tab-panel">
                This is the tab content.
              </ssr-keep>
            </div>
          </div>
        </bolt-tab-panel>
      </ssr-keep>
    </div>
    <input type="radio"
           name="tabs-12345"
           id="tabs-12345-3"
           class="c-bolt-tabs__input"
    >
    <label for="tabs-12345-3"
           class="c-bolt-tabs__label c-bolt-tabs__label--spacing-small"
    >
      <span class="c-bolt-tabs__label-inner">
        <span class="c-bolt-tabs__label-text">
          Tab label 3
        </span>
      </span>
    </label>
    <div class="c-bolt-tab-panel__outer">
      <ssr-keep for="bolt-tabs">
        <bolt-tab-panel>
          <ssr-keep for="bolt-tab-panel">
            <div slot="label">
              Tab label 3
            </div>
          </ssr-keep>
          <div class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small">
            <div role="tabpanel"
                 class="c-bolt-tab-panel__content"
            >
              <ssr-keep for="bolt-tab-panel">
                This is the tab content.
              </ssr-keep>
            </div>
          </div>
        </bolt-tab-panel>
      </ssr-keep>
    </div>
  </div>
</bolt-tabs>
`;

exports[`Bolt Tabs Web Component usage (Light DOM) 2`] = `
<bolt-tabs
  no-shadow=""
  ready=""
  style=""
>
  <!---->
  
       
      
  <div
    class="c-bolt-tabs c-bolt-tabs--align-start c-bolt-tabs--inset"
    style=""
  >
    
        
    <div
      class="c-bolt-tabs__nav"
      role="tablist"
      style=""
    >
      
          
      <!---->
      
          
      <bolt-trigger
        aria-controls="tab-panel-12345-1"
        aria-selected="true"
        class="c-bolt-tabs__label c-bolt-tabs__item c-bolt-tabs__label--spacing-small"
        display="inline"
        id="tab-label-12345-1"
        no-outline=""
        role="tab"
        style=""
        tabindex="0"
      >
        
            
        <span
          class="c-bolt-tabs__label-inner"
          style=""
        >
          <span
            class="c-bolt-tabs__label-text"
            style=""
          >
            <!---->
            Tab label 1
            <!---->
          </span>
        </span>
        
          
      </bolt-trigger>
      
        
      <!---->
      
          
      <bolt-trigger
        aria-controls="tab-panel-12345-2"
        aria-selected="false"
        class="c-bolt-tabs__label c-bolt-tabs__item c-bolt-tabs__label--spacing-small"
        display="inline"
        id="tab-label-12345-2"
        no-outline=""
        role="tab"
        style=""
        tabindex="-1"
      >
        
            
        <span
          class="c-bolt-tabs__label-inner"
          style=""
        >
          <span
            class="c-bolt-tabs__label-text"
            style=""
          >
            <!---->
            Tab label 2
            <!---->
          </span>
        </span>
        
          
      </bolt-trigger>
      
        
      <!---->
      
          
      <bolt-trigger
        aria-controls="tab-panel-12345-3"
        aria-selected="false"
        class="c-bolt-tabs__label c-bolt-tabs__item c-bolt-tabs__label--spacing-small"
        display="inline"
        id="tab-label-12345-3"
        no-outline=""
        role="tab"
        style=""
        tabindex="-1"
      >
        
            
        <span
          class="c-bolt-tabs__label-inner"
          style=""
        >
          <span
            class="c-bolt-tabs__label-text"
            style=""
          >
            <!---->
            Tab label 3
            <!---->
          </span>
        </span>
        
          
      </bolt-trigger>
      
        
      <!---->
       
        
      <div
        class="c-bolt-tabs__item c-bolt-tabs__show-more"
        style=""
      >
        
          
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="c-bolt-tabs__button c-bolt-tabs__show-button"
          style=""
          type="button"
        >
          
            
          <span
            class="c-bolt-tabs__show-text"
            style=""
          >
            
              More
            
          </span>
          
            
          <span
            class="c-bolt-tabs__show-icon"
            style=""
          >
            
              
            <bolt-icon
              name="chevron-down"
              style=""
            />
            
            
          </span>
          
          
        </button>
        
          
        <div
          class="c-bolt-tabs__dropdown"
          style=""
        >
          
            
          <div
            class="c-bolt-tabs__dropdown-list"
            style=""
          >
            
              
            <!---->
            
          
            <bolt-trigger
              aria-controls="tab-panel-12345-1"
              aria-selected="true"
              class="c-bolt-tabs__label c-bolt-tabs__item c-bolt-tabs__label--spacing-small c-bolt-tabs__label--is-duplicate c-bolt-tabs__label--vertical-border"
              display="block"
              id="tab-dropdown-12345-1"
              no-outline=""
              role="tab"
              style=""
              tabindex="0"
            >
              
            
              <span
                class="c-bolt-tabs__label-inner"
                style=""
              >
                <span
                  class="c-bolt-tabs__label-text"
                  style=""
                >
                  <!---->
                  Tab label 1
                  <!---->
                </span>
              </span>
              
          
            </bolt-trigger>
            
        
            <!---->
            
          
            <bolt-trigger
              aria-controls="tab-panel-12345-2"
              aria-selected="false"
              class="c-bolt-tabs__label c-bolt-tabs__item c-bolt-tabs__label--spacing-small c-bolt-tabs__label--is-duplicate c-bolt-tabs__label--vertical-border"
              display="block"
              id="tab-dropdown-12345-2"
              no-outline=""
              role="tab"
              style=""
              tabindex="-1"
            >
              
            
              <span
                class="c-bolt-tabs__label-inner"
                style=""
              >
                <span
                  class="c-bolt-tabs__label-text"
                  style=""
                >
                  <!---->
                  Tab label 2
                  <!---->
                </span>
              </span>
              
          
            </bolt-trigger>
            
        
            <!---->
            
          
            <bolt-trigger
              aria-controls="tab-panel-12345-3"
              aria-selected="false"
              class="c-bolt-tabs__label c-bolt-tabs__item c-bolt-tabs__label--spacing-small c-bolt-tabs__label--is-duplicate c-bolt-tabs__label--vertical-border"
              display="block"
              id="tab-dropdown-12345-3"
              no-outline=""
              role="tab"
              style=""
              tabindex="-1"
            >
              
            
              <span
                class="c-bolt-tabs__label-inner"
                style=""
              >
                <span
                  class="c-bolt-tabs__label-text"
                  style=""
                >
                  <!---->
                  Tab label 3
                  <!---->
                </span>
              </span>
              
          
            </bolt-trigger>
            
        
            <!---->
            
            
          </div>
          
          
        </div>
        
        
      </div>
      
      
        
    </div>
    
        
    <div
      class="c-bolt-tabs__panels-container"
      style=""
    >
      
          
      <!---->
      <!---->
      
    
      <!---->
      <bolt-tab-panel
        no-shadow=""
        selected=""
        style=""
      >
        <!---->
        
       
      
        <div
          class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small"
          style=""
        >
          <!---->
          <!---->
          <!---->
          <!---->
          <div
            slot="label"
            style=""
          >
            Tab label 1
          </div>
          <!---->
          <!---->
          <!---->
          
                
          <div
            aria-expanded="true"
            aria-labelledby="tab-label-12345-1"
            class="c-bolt-tab-panel__content"
            id="tab-panel-12345-1"
            role="tabpanel"
            style=""
            tabindex="0"
          >
            
                  
            <!---->
            <!---->
            
      
            <!---->
            
      Tab panel 1
    
            <!---->
            <!---->
            
                
          </div>
          
              
          <!---->
          <!---->
        </div>
        
    
    
        <!---->
      </bolt-tab-panel>
      <!---->
      
    
      <!---->
      <bolt-tab-panel
        no-shadow=""
        style=""
      >
        <!---->
        
       
      
        <div
          class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small"
          style=""
        >
          <!---->
          <!---->
          <!---->
          <!---->
          <div
            slot="label"
            style=""
          >
            Tab label 2
          </div>
          <!---->
          <!---->
          <!---->
          
                
          <div
            aria-expanded="false"
            aria-labelledby="tab-label-12345-2"
            class="c-bolt-tab-panel__content"
            id="tab-panel-12345-2"
            role="tabpanel"
            style=""
            tabindex="0"
          >
            
                  
            <!---->
            <!---->
            
      
            <!---->
            
      Tab panel 2
    
            <!---->
            <!---->
            
                
          </div>
          
              
          <!---->
          <!---->
        </div>
        
    
    
        <!---->
      </bolt-tab-panel>
      <!---->
      
    
      <!---->
      <bolt-tab-panel
        no-shadow=""
        style=""
      >
        <!---->
        
       
      
        <div
          class="c-bolt-tab-panel c-bolt-tab-panel--spacing-small"
          style=""
        >
          <!---->
          <!---->
          <!---->
          <!---->
          <div
            slot="label"
            style=""
          >
            Tab label 3
          </div>
          <!---->
          <!---->
          <!---->
          
                
          <div
            aria-expanded="false"
            aria-labelledby="tab-label-12345-3"
            class="c-bolt-tab-panel__content"
            id="tab-panel-12345-3"
            role="tabpanel"
            style=""
            tabindex="0"
          >
            
                  
            <!---->
            <!---->
            
      
            <!---->
            
      Tab panel 3
    
            <!---->
            <!---->
            
                
          </div>
          
              
          <!---->
          <!---->
        </div>
        
    
    
        <!---->
      </bolt-tab-panel>
      <!---->
      
  
      <!---->
      <!---->
      
        
    </div>
    
      
  </div>
  
    
    
  <!---->
</bolt-tabs>
`;

exports[`Bolt Tabs Web Component usage (Shadow DOM) 2`] = `
<bolt-tabs
  ready=""
  style=""
>
  
    
  <bolt-tab-panel
    selected=""
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 1
    </div>
    
      Tab panel 1
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 2
    </div>
    
      Tab panel 2
    
  </bolt-tab-panel>
  
    
  <bolt-tab-panel
    style=""
  >
    
      
    <div
      slot="label"
      style=""
    >
      Tab label 3
    </div>
    
      Tab panel 3
    
  </bolt-tab-panel>
  
  
</bolt-tabs>
`;
